plugins/mcreator-core/blockly/field_data_list_selector.js

```diff
@@ -0,0 +1,126 @@
+ * The behaviour is similar to block/item selectors or condition selectors for entity AI blocks
+ */
+function FieldDataListSelector(type) {
+    // The default entry is ",No entry selected". Since the value is an empty string, the procedure editor will show a compile error
+    let getDefaultEntry = function () {
+        return ',' + javabridge.t('blockly.extension.data_list_selector.no_entry');
+    }
+
+    // While the procedure is open, we store the selected entry as a "value,readableName" pair
+    let entry = getDefaultEntry();
+
+    // The clickable part of the custom field
+    let entryField = new Blockly.FieldLabelSerializable(javabridge.t('blockly.extension.data_list_selector.no_entry'), 'entry-label');
+    entryField.EDITABLE = true;
+    entryField.SERIALIZABLE = true;
+
+    // Initialize the label with a rectangle surrounding the text
+    entryField.initView = function () {
+        let rect = Blockly.utils.dom.createSvgElement('rect',
+            {
+                'class': 'blocklyFlyoutButtonShadow',
+                'rx': 2, 'ry': 2, 'y': 0, 'x': 1
+            },
+            this.fieldGroup_);
+
+        this.createTextElement_();
+
+        if (workspace.getRenderer().name === "thrasos") {
+            this.textElement_.setAttribute("y", 8);
+            this.textElement_.setAttribute("x", this.textElement_.getAttribute("x") + 4);
+        } else {
+            this.textElement_.setAttribute("y", 13);
+            this.textElement_.setAttribute("x", this.textElement_.getAttribute("x") + 5);
+        }
+
+        if (this.class_)
+            Blockly.utils.dom.addClass(this.textElement_, this.class_);
+
+        rect.setAttribute('width', 93);
+        rect.setAttribute('height', 15);
+        this.rect = rect; // This is so we can update its shape
+
+        this.lastClickTime = -1;
+    };
+
+    // Updates the shape of the field and of the rectangle surrounding the text
+    entryField.updateSize_ = function () {
+        this.size_.height = 14;
+        if (this.textElement_)
+            this.size_.width = Blockly.utils.dom.getTextWidth(this.textElement_) + 12;
+        else
+            this.size_.width = 93;
+        this.rect.setAttribute('width', Blockly.utils.dom.getTextWidth(this.textElement_) + 8);
+    };
+
+    // Function to handle clicking
+    entryField.onMouseDown_ = function (e) {
+        if (this.sourceBlock_ && !this.sourceBlock_.isInFlyout) {
+            if (this.lastClickTime !== -1 && ((new Date().getTime() - this.lastClickTime) < 500)) {
+                e.stopPropagation(); // fix so the block does not "stick" to the mouse when the field is clicked
+                javabridge.openEntrySelector(type, {
+                    'callback': function (data) {
+                        if (data !== undefined) {
+                            entry = data;
+                        } else {
+                            entry = getDefaultEntry();

```

Comment:
You could _maybe_ provide close button response here, like [I did](https://github.com/MCreator/MCreator/pull/2000/commits/f2406b0b2e18c3ad0be3f6a829a811713ffd628c#diff-a80b078a5c13fb128333e36fe91cb1f505227dedc144a9a74cb1b1785103720bR41-R42) to the state editor.

Reply:
Block selector also does not have close so I think this is ok, there is no close, X actually clears the field